lib/app-components/addon/components/branded-navbar/component.ts

```diff
@@ -0,0 +1,42 @@
+import { tagName } from '@ember-decorators/component';
+import { computed } from '@ember-decorators/object';
+import { service } from '@ember-decorators/service';
+import Component from '@ember/component';
+import I18N from 'ember-i18n/services/i18n';
+import { serviceLinks } from 'ember-osf-web/const/service-links';
+import requiredAction from 'ember-osf-web/decorators/required-action';
+import Analytics from 'ember-osf-web/services/analytics';
+import Theme from 'ember-osf-web/services/theme';
+import Session from 'ember-simple-auth/services/session';
+import styles from './styles';
+import layout from './template';
+
+type ObjectType = 'collection' | 'preprint' | 'registration';
+
+@tagName('')
+export default class BrandedNavbar extends Component {
+    layout = layout;
+    styles = styles;
+
+    @service analytics!: Analytics;
+    @service i18n!: I18N;
+    @service session!: Session;
+    @service theme!: Theme;
+
+    brandRoute: string = this.brandRoute;
+    objectType: ObjectType = this.objectType;
+    signupUrl: string = this.signupUrl;
+
+    myProjectsUrl = serviceLinks.myProjects;
+
+    key = 'collections.navbar.brand';
+
+    @computed('i18n.locale', 'theme.provider')
```

Comment:
I guess it doesn't hurt to have this depend on `i18n.locale`, but do we expect that to ever change after the app initializes?